### PR TITLE
Update sidebar items and style

### PR DESCRIPTION
**Pull request by vivienherq**
# Pull Request

## Description
Fixed the problem of the sidebar scrolling with page and added back the sidebar items (assignments, submissions, log out).

## Screenshots (if applicable)

Collapsed:
Screenshot 2024-03-18 at 2 56 18 AM

Expanded:
Screenshot 2024-03-18 at 2 56 44 AM

## Checklist

- [ ] I have checked that the changes included in the PR are intended to merge to `master` or any destination branch.
- [ ] I have verified that the new changes do not break any existing functionalities, unless the new changes are intended and have approved by the team.
- [ ] I will take care of the merging and delete the side-branch after the PR is merged.

## Additional Notes/References

I saw a previous commit fixing the sidebar but I think it made the sidebar scrollable, this sidebar should be the full height of the screen and shouldn't be scrollable.

### CODE REVIEW

File: frontend/src/components/common/SideBar.tsx [hunk 1]
```diff
@@ -45,31 +66,75 @@ const SideBar = () => {
...
                 onClick={handleToggleCollapse}
-                className="text-black shadow-lg"
+                className="text-black"
               >
                 <HiMenu className="text-2xl" />
               </Button>
-              <Avatar src="https://i.pravatar.cc/150?u=a042581f4e29026024d" />
+              <Avatar src="https://i.pravatar.cc/150?u=a04258114e29026702d" />
+              <Spacer y={52} />
+              {menuItems.map((item: any) => (
```
Comment (tryyang2001) [inline frontend/src/components/common/SideBar.tsx:75]: Should have a `key` property here to prevent the error in the browser

**Comment by tryyang2001** [comment]
Hmm, I tested on my local comp with 16 inches screen, the assignment page doesn't seem to render well with sidebar:
image

**Comment by vivienherq** [comment]
> Hmm, I tested on my local comp with 16 inches screen, the assignment page doesn't seem to render well with sidebar: image

could you try again! i think i managed to fix the layout issue causing this
Screenshot 2024-03-18 at 12 43 10 PM
Screenshot 2024-03-18 at 12 43 22 PM

**Comment by tryyang2001** [comment]
Ok yes the render issue has been fixed, but I notice that when I first run `yarn dev`, the avatar in the sidebar will not show up and none of the links in the sidebar are clickable, do you have any idea what causes this? See the screenshot below for example:
image

**Comment by tryyang2001** [comment]
Ah and also I am a bit confused for the build error, can you try pulling from master branch to fetch the latest changes that will resolve the error and see if there are any build error?

**Comment by vivienherq** [comment]
> Ah and also I am a bit confused for the build error, can you try pulling from master branch to fetch the latest changes that will resolve the error and see if there are any build error?

after pulling from main, i'm getting error when i try to `yarn dev`, do you know how to fix this?

Screenshot 2024-03-18 at 11 11 25 AM

**Comment by vivienherq** [comment]
> Ah and also I am a bit confused for the build error, can you try pulling from master branch to fetch the latest changes that will resolve the error and see if there are any build error?

there aren't any build errors on my side after running `yarn build`... could you post a screenshot of the error?

**Comment by vivienherq** [comment]
> Ok yes the render issue has been fixed, but I notice that when I first run `yarn dev`, the avatar in the sidebar will not show up and none of the links in the sidebar are clickable, do you have any idea what causes this? See the screenshot below for example: image A refresh will solve this problem, but not sure what happens tho... (perhaps you can try using useEffect to ensure you have retrieve the avatar before rendering?)

with regards to the buttons, can you check if the url has changed after clicking on the buttons? because they should bring you to assignments/dashboard and assignments/submissions respectively, but since the PR containing these hasn't been merged yet, so currently both urls are still bringing you to the assignment page.

also i'm not facing the avatar issue (for me it loads without having to refresh the page) so i'm not really sure what's wrong with that...

**Comment by tryyang2001** [comment]
Ok great, there is no more build error on my side as well. But the problem I mentioned above regarding the avatar cannot be rendered in the sidebar on the first run still persists, and it only occurs in dev (when I run `yarn dev`), and not in production (`yarn build` follow by `yarn start`).

**Comment by tryyang2001** [comment]
Ok the problem is something with the React `strictMode`, we can either add `reactStrictMode: false` to `next.config.mjs`, or simply ignore this.

**Review by tryyang2001 — APPROVED** [review]

**Comment by vivienherq** [comment]
> Ok the problem is something with the React `strictMode`, we can either add `reactStrictMode: false` to `next.config.mjs`, or simply ignore this.

hmm, for now i added a fallback, so the avatar should load with the user's name before the image is done loading

**Pull request merged**